src/main/java/at/ac/tuwien/kr/alpha/solver/learning/GroundConflictNoGoodLearner.java

```diff
@@ -54,7 +54,26 @@
 	private final Assignment assignment;
 	private final AtomStore atomStore;
 
+	/**
+	 * Given a conflicting NoGood, computes a conflict-free backjumping level such that the given NoGood is not
+	 * violated.
+	 *
+	 * @param violatedNoGood the conflicting NoGood.
+	 * @return -1 if the violatedNoGood cannot be satisfied, otherwise
+	 * 	   0 if violatedNoGood is unary, and else
+	 * 	   the highest backjumping level such that the NoGood is no longer violated.
+	 */
 	public int computeConflictFreeBackjumpingLevel(NoGood violatedNoGood) {
+		// If violatedNoGood is unary, backjump to decision level 0 if it can be satisfied.
+		if (violatedNoGood.size() == 1) {
+			int literal = violatedNoGood.getLiteral(0);
+			int literalDecisionLevel = assignment.getRealWeakDecisionLevel(atomOf(literal));
+			if (literalDecisionLevel > 0) {
+				return 0;
```

Comment:
That is indeed a good question and it is not fully clear to me, what is the optimal choice.

In favour of `DL = 0` we have:
- We know the unary nogood will be assigning at 0.
- 0 is the lowest DL where the nogood is assigning, if we backjump less, we have at least one out-of-order literal, potentially even more if that triggers other implications.
- If phase saving is enabled (which in SAT solvers is usually the case), then this amounts to a "soft restart" and the phase saving will lead the solver directly to the same spot in the search space it has been before (but now some decision levels are better). So the backjump to 0 should not really have any significant drawbacks (at least for SAT solvers -- for Alpha it is not so clear due to the restricted guessing).

In favour of `literalDecisionLevel - 1` we have:
- It is the smallest possible backjump.
- It will not create the potential drawbacks of a restart but potentially induce more work with out-of-order literals and also propagations on higher decision levels than actually needed.

So there are no hard facts for deciding either way. Generally, it is better to have assignments on lower decision levels and since we have a PR enabling phase saving, I decided for `DL=0`.